exchanges/orderbook/depth.go

```diff
@@ -138,7 +147,14 @@ func (d *Depth) IsValid() bool {
 
 // UpdateBidAskByPrice updates the bid and ask spread by supplied updates, this
 // will trim total length of depth level to a specified supplied number
-func (d *Depth) UpdateBidAskByPrice(update *Update) {
+func (d *Depth) UpdateBidAskByPrice(update *Update) error {
+	if update.UpdateTime.IsZero() {
+		return fmt.Errorf("%s %s %s %w",
+			d.exchange,
+			d.pair,
+			d.asset,
+			errLastUpdatedNotSet)
+	}
 	tn := getNow()
```

Comment:
Why do we use this when we now have an updateTime?

Reply:
Nar not really, Last updated is more a time for when the update occurred at a matching engine level on the external exchange. This time is for the stack nodes when it gets pushed onto the stack to be re-used later more like an insertion time so it can be clean if its never been used. Either way the times are going to be like a second or two out depending on colocation from exchange and processing time. Default allowance is 30 seconds so might as well use it and reduce allocation. 🤷 🦅